userspace/libsinsp/threadinfo.cpp

```diff
@@ -517,19 +517,27 @@ void sinsp_threadinfo::set_user(uint32_t uid)
 	if (!user)
 	{
 		auto notify = m_inspector->is_live() || m_inspector->is_syscall_plugin();
-		user = m_inspector->m_usergroup_manager.add_user(m_container_id, m_pid, uid, m_group.gid, {}, {}, {}, notify);
+		user = m_inspector->m_usergroup_manager.add_user(m_container_id, m_pid, uid, m_group->gid(), {}, {}, {}, notify);
 	}
+
 	if (user)
 	{
-		memcpy(&m_user, user, sizeof(scap_userinfo));
+		m_user->set_uid(user->uid);
+		m_user->set_gid(m_group->gid());
+
+		if (m_inspector->is_user_details_enabled())
+		{
+			m_user->set_name(user->name, sizeof(user->name));
+			m_user->set_homedir(user->homedir, sizeof(user->homedir));
+			m_user->set_shell(user->shell, sizeof(user->shell));
+		}
 	}
 	else
 	{
-		m_user.uid = uid;
-		m_user.gid = m_group.gid;
-		strlcpy(m_user.name, (uid == 0) ? "root" : "<NA>", sizeof(m_user.name));
-		strlcpy(m_user.homedir, (uid == 0) ? "/root" : "<NA>", sizeof(m_user.homedir));
-		strlcpy(m_user.shell, "<NA>", sizeof(m_user.shell));
+		// No need to set name/homedir/shell, the default values from
+		// sinsp_userinfo are going to be used.
+		m_user->set_uid(uid);
```

Comment:
Actually,
```
strlcpy(m_user.name, (uid == 0) ? "root" : "<NA>", sizeof(m_user.name));
strlcpy(m_user.homedir, (uid == 0) ? "/root" : "<NA>", sizeof(m_user.homedir));
```
was a workaround for when we got uid 0 but we don't have any matching user. If i remeber correctly, this can happen during early stage for containers. cc @deepskyblue86

Reply:
I think we should maintain the behavior.